src/main/java/com/palantir/docker/compose/execution/AbstractCommand.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.docker.compose.execution;
+
+import static java.util.stream.Collectors.joining;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public abstract class AbstractCommand<T extends Executable> {
+
+    private final String commandName;
+    private final AbstractSynchronousExecutable<T> executable;
+
+    public AbstractCommand(String commandName, AbstractSynchronousExecutable<T> executable) {
```

Comment:
Having the command name here results in duplication between the Executable classes (which know the name of the command they are executing as a requirement of performing their responsibility) and the children of this class. Instead, how about adding another method to the executable interface that provides this information for the purpose of constructing error messages.